hail_search/queries/mito.py

```diff
@@ -205,37 +205,61 @@ def _filter_entries_ht_both_sample_types(
             )
 
         ch_ht = None
-        family_guid_idx_map = defaultdict(dict)
+        family_idx_map = defaultdict(dict)
         for sample_type, sorted_family_sample_data in sample_types:
+            ht = self._annotate_empty_failed_inheritance(ht, sample_type)
+            ch_ht = self._annotate_empty_failed_inheritance(ch_ht, sample_type)
+
             ht, ch_ht = self._filter_inheritance(
                 ht, ch_ht, inheritance_filter, sorted_family_sample_data,
-                annotation=sample_type.passes_inheritance_field, entries_ht_field=sample_type.family_entries_field
+                annotation=sample_type.failed_family_sample_field, entries_ht_field=sample_type.family_entries_field,
+                family_passes_inheritance_filter=self._get_family_passes_inheritance_filter_both_sample_types
             )
             for family_idx, samples in enumerate(sorted_family_sample_data):
                 family_guid = samples[0]['familyGuid']
-                family_guid_idx_map[family_guid][sample_type.value] = family_idx
+                family_idx_map[family_guid][sample_type.value] = family_idx
 
-        family_idx_map = hl.dict(family_guid_idx_map)
+        family_idx_map = hl.dict(family_idx_map)
         ht = self._apply_multi_sample_type_entry_filters(ht, family_idx_map)
         ch_ht = self._apply_multi_sample_type_entry_filters(ch_ht, family_idx_map)
         return ht, ch_ht
 
+    @staticmethod
+    def _annotate_empty_failed_inheritance(ht, sample_type):
+        if ht is None:
+            return ht
+
+        return ht.annotate(**{
+            sample_type.failed_family_sample_field: ht[sample_type.family_entries_field].map(
+                lambda x: hl.empty_array(hl.tstr)
+            )})
+
+    def _get_family_passes_inheritance_filter_both_sample_types(
+        self, entry_indices, family_idx, genotype, family_samples, ht, annotation
+    ):
+        return hl.bind(
+            lambda failed_samples: ht[annotation][family_idx].extend(failed_samples),
+            entry_indices.get(family_idx).filter(
+                lambda sample_idx: ~self.GENOTYPE_QUERY_MAP[genotype](family_samples[sample_idx].GT)
+            ).map(lambda sample_idx: family_samples[sample_idx]['sampleId'])
+        )
```

Comment:
Bind is actually less readable and does not have any performance boost if the bound variable is used exactly once and not within a loop, like in this case
```suggestion
        return ht[annotation][family_idx].extend(
            entry_indices.get(family_idx).filter(
                lambda sample_idx: ~self.GENOTYPE_QUERY_MAP[genotype](family_samples[sample_idx].GT)
            ).map(lambda sample_idx: family_samples[sample_idx]['sampleId'])
        ))
```

Reply:
resolving this because I changed the structure of the annotation so this comment no longer applies. I also didn't use a bind here in my most recent refactor.